app/src/main/java/org/elasticsearch/node/internal/CrateSettingsPreparer.java

```diff
@@ -150,17 +150,21 @@ private static String randomNodeName() {
 
         try {
             List<String> names = new ArrayList<>();
-            try (BufferedReader reader = new BufferedReader(new InputStreamReader(input, Charsets.UTF_8))) {
-                String line = reader.readLine();
-                while (line != null) {
-                    String[] fields = line.split("\t");
-                    if (fields.length == 0) {
-                        throw new RuntimeException("Failed to parse the names.txt. Malformed record: " + line);
+            try (InputStreamReader streamReader = new InputStreamReader(input, Charsets.UTF_8)) {
+                try (BufferedReader reader = new BufferedReader(streamReader)) {
+                    String line = reader.readLine();
+                    while (line != null) {
+                        String[] fields = line.split("\t");
+                        if (fields.length == 0) {
+                            input.close();
+                            throw new RuntimeException("Failed to parse the names.txt. Malformed record: " + line);
+                        }
+                        names.add(fields[0]);
+                        line = reader.readLine();
                     }
-                    names.add(fields[0]);
-                    line = reader.readLine();
                 }
             }
+            input.close();
```

Comment:
Shouldn't this go into a `finally` block to be safe?

Reply:
close() can also raise an IOException
that's why I put it into the try block and also close in case of a malformed line